test/grpc/route_guide/route-guide-eventuals-test.cc

```diff
@@ -0,0 +1,64 @@
+#include "route-guide-eventuals-test.h"
+
+#include <grpcpp/channel.h>
+
+#include "route-guide-eventuals-server.h"
+#include "test/grpc/route_guide/helper.h"
+#include "test/grpc/route_guide/route_guide.eventuals.h"
+#include "test/grpc/route_guide/route_guide.grpc.pb.h"
+
+using eventuals::grpc::ServerBuilder;
+
+////////////////////////////////////////////////////////////////////////
+
+void RouteGuideImpl::ParseDb(const std::string& db) {
+  routeguide::ParseDb(db, &feature_list_);
+}
+
+Feature RouteGuideImpl::GetFeature(
+    grpc::ServerContext* context,
+    Point&& point) {
+  Feature feature;
+  feature.set_name(GetFeatureName(point, feature_list_));
+  feature.mutable_location()->CopyFrom(point);
+  return feature;
+}
+
+////////////////////////////////////////////////////////////////////////
+
+int RouteGuideTest::GetPointsCount() {
+  return kPoints_;
+}
+
+void RouteGuideTest::SetDb(const std::string& db) {
+  routeguide::ParseDb(db, &feature_list_);
+}
+
+void RouteGuideTest::SetUp() {
```

Comment:
Let's use domain sockets instead of having to deal with ports. That way you can construct a `Client` before the server and it doesn't have to be a `std::optional`. You'll need to use a `TemporaryDirectoryTest`. See example in `test/grpc/client-interruptible.cc`